function/index.js

```diff
@@ -12,7 +12,8 @@ const client = new automl.PredictionServiceClient();
  * @param {Object} res Cloud Function response context.
  *                     More info: https://expressjs.com/en/api.html#res
  */
-exports.predictImage = (req, res) => {
+
+@@ -19,14 +19,29 @@ exports.predictImage = (req, res) => {
```

Comment:
Right after this line, add:

```
if (req.method === "OPTIONS") {
      res.set("Access-Control-Allow-Origin", "*")
      res.set("Access-Control-Allow-Headers", "Accept, Accept-Language, Content-Language, Content-Type")
      res.status(200).send()
    }
  
  	if (req.method !== "POST") {
      res.status(405).send()
    }
```

The browser needs this to send requests to a domain (the Cloud Functions one) different from where it’s hosted. 

Copied from the existing function: https://console.cloud.google.com/functions/details/us-central1/predict?project=sp19-codeu-35-7727&tab=source&duration=PT1H